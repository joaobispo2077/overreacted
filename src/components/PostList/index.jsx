import React from 'react'
import { PostCard } from '../PostCard'
import './styles.css';

export default function PostList({ posts = [] }) {
  return (
    <ul className="posts">
      {posts.map(post =>
        <PostCard
          key={post.id}
          title={post.title}
          body={post.body}
          id={post.id}
          cover={post.cover}
        />
      )}
    </ul>
  )
}
