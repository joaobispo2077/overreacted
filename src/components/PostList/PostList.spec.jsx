import { render, screen } from "@testing-library/react";
import PostList from '.';

const createMockPosts = () => {
  const mockPosts = [
    {
      id: 1,
      title: 'title 1',
      body: 'body 1',
      cover: 'img/img1.png'
    },
    {
      id: 2,
      title: 'title 2',
      body: 'body 2',
      cover: 'img/img2.png'
    },
    {
      id: 3,
      title: 'title 3',
      body: 'body 3',
      cover: 'img/img3.png'
    },
  ];

  return mockPosts;
};


describe('<PostList />', () => {
  it('should render posts', () => {
    const mockPosts = createMockPosts();
    render(<PostList posts={mockPosts} />);

    expect(screen.getAllByRole('heading', { name: /title/i })).toHaveLength(3);
    expect(screen.getAllByRole('img', { name: /title/i })).toHaveLength(3);
    expect(screen.getAllByText(/title/i)).toHaveLength(3);
    expect(screen.getByRole('img', { name: /title 3/i })).toHaveAttribute('src', 'img/img3.png');
  });

  it('should match snapshot', () => {
    const mockPosts = createMockPosts();
    const { container } = render(<PostList posts={mockPosts} />);

    expect(container.firstChild).toMatchSnapshot();
  });
});