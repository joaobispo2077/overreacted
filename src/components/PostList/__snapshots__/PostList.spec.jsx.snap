// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<PostList /> should match snapshot 1`] = `
<ul
  class="posts"
>
  <li
    class="post"
  >
    <img
      alt="title 1"
      src="img/img1.png"
    />
    <div
      class="post-content"
    >
      <h1>
        1
         - 
        title 1
      </h1>
      <p>
        body 1
      </p>
    </div>
  </li>
  <li
    class="post"
  >
    <img
      alt="title 2"
      src="img/img2.png"
    />
    <div
      class="post-content"
    >
      <h1>
        2
         - 
        title 2
      </h1>
      <p>
        body 2
      </p>
    </div>
  </li>
  <li
    class="post"
  >
    <img
      alt="title 3"
      src="img/img3.png"
    />
    <div
      class="post-content"
    >
      <h1>
        3
         - 
        title 3
      </h1>
      <p>
        body 3
      </p>
    </div>
  </li>
</ul>
`;
